packages/across-plugin/src/service.ts

```diff
@@ -0,0 +1,405 @@
+      this.getSuggestedFees({
+        originChainId: parseInt(route.source.chainId),
+        destinationChainId: parseInt(route.destination.chainId),
+        token: route.source.assetId,
+        amount: baselineAmount,
+      })
+    );
+
+    const baselineFeePct = parseFloat(baselineQuote.totalRelayFee.pct);
+
+    // Find amounts that meet slippage thresholds
+    let threshold50: string = testAmounts[0];
+    let threshold100: string = testAmounts[0];
+
+    for (const amount of testAmounts) {
+      try {
+        const quote = await this.rateLimiter.schedule(() =>
+          this.getSuggestedFees({
+            originChainId: parseInt(route.source.chainId),
+            destinationChainId: parseInt(route.destination.chainId),
+            token: route.source.assetId,
+            amount,
+          })
+        );
+
+        const currentFeePct = parseFloat(quote.totalRelayFee.pct);
+        const slippageBps = Math.abs((currentFeePct - baselineFeePct) * 10000);
+
+        if (slippageBps <= 50) {
+          threshold50 = amount;
+        }
+        if (slippageBps <= 100) {
+          threshold100 = amount;
+        }
+      } catch (error) {
+        console.error(`[AcrossDataProvider] Failed to test amount ${amount}:`, error);
+        break;
+      }
+    }
+
+    return [
+      { maxAmountIn: threshold50, slippageBps: 50 },
+      { maxAmountIn: threshold100, slippageBps: 100 },
+    ];
+  }
+
+  
+  private async getListedAssets(): Promise<ListedAssetsType> {
+    console.log(`[AcrossDataProvider] Fetching available assets`);
+
+    const routes = await this.getAvailableRoutes();
+
+    // Extract unique assets from routes
+    const assetsMap = new Map<string, AssetType>();
+
+    for (const route of routes) {
+      // Add source asset
+      const sourceKey = `${route.originChainId}-${route.originToken}`;
+      if (!assetsMap.has(sourceKey)) {
+        assetsMap.set(sourceKey, {
+          chainId: route.originChainId.toString(),
+          assetId: route.originToken,
+          symbol: route.originTokenSymbol,
+          decimals: this.getTokenDecimals(route.originTokenSymbol),
+        });
+      }
+
+      // Add destination asset
+      const destKey = `${route.destinationChainId}-${route.destinationToken}`;
+      if (!assetsMap.has(destKey)) {
+        assetsMap.set(destKey, {
+          chainId: route.destinationChainId.toString(),
+          assetId: route.destinationToken,
+          symbol: route.destinationTokenSymbol,
+          decimals: this.getTokenDecimals(route.destinationTokenSymbol),
+        });
+      }
+    }
+
+    const assets = Array.from(assetsMap.values());
+    console.log(`[AcrossDataProvider] Found ${assets.length} unique assets`);
+
+    return {
+      assets,
+      measuredAt: new Date().toISOString(),
+    };
+  }
+
+  
+  private async getAvailableRoutes(): Promise<AcrossRoute[]> {
+    const now = Date.now();
+    if (this.routesCache && now - this.routesCacheTime < this.CACHE_TTL) {
```

Comment:
Do we need these to be cached? What's the justification? Were you hitting rate limits?